### PR TITLE
Terraform prow static ip

### PR DESCRIPTION
**Pull request by inteon**
Thanks to @maelvls for the tip wrt. the annotation.

Corresponds to this prow change PR: https://github.com/cert-manager/testing/pull/1022

### CODE REVIEW

File: gcp/modules/gcp-cluster/variables.tf [hunk 1]
```diff
@@ -24,6 +24,11 @@ variable "cluster_enable_workload_identity" {
   type        = bool
   default     = false
 }
+variable "cluster_enable_http_load_balancing" {
+  description = "Whether to enable the HTTP load balancing addon"
+  type        = bool
+  default     = false
+}
```
Comment (maelvls) [inline gcp/modules/gcp-cluster/variables.tf:31]: Why is this needed?

**Comment by maelvls** [comment]
For future reference, here is [the tip](https://venafi.slack.com/archives/C054TTTN7S6/p1714733430013519) Tim is referring to:

Regarding Prow's IP being created "from Kubernetes" with `type:LoadBalancer` and then manually copied to the Terraform configuration, there is this annotation ([docs](https://cloud.google.com/kubernetes-engine/docs/concepts/ingress-xlb#static_ip_addresses_for_https_load_balancers)):

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: my-ingress
  annotations:
    kubernetes.io/ingress.global-static-ip-name: my-static-address
```

The idea would be:

1. Terraform creates a static IP with a given name,
2. The Ingress manifest points to that name in Kubernetes.

Gateway API [also supports](https://cloud.google.com/kubernetes-engine/docs/how-to/deploying-gateways) the same mechanism if needed (if we ever feel the need to migrate...)

File: gcp/modules/gcp-cluster/variables.tf (hunk 1 above)
Comment (inteon) [reply gcp/modules/gcp-cluster/variables.tf:31]: The `kubernetes.io/ingress.global-static-ip-name: prow-infra-cert-manager-io` annotation is only supported by the GCP ingress controller. This option basically enables that controller. Before, we were using nginx instead.

Comment (inteon) [reply gcp/modules/gcp-cluster/variables.tf:31]: https://github.com/kubernetes/ingress-gce

Comment (maelvls) [reply gcp/modules/gcp-cluster/variables.tf:31]: Ah, I didn't see that `http_load_balancing` was previously set to "disabled". So `http_load_balancing` corresponds to ingress-gce, which we previously didn't use. Got it.

**Review by maelvls — APPROVED** [review]
